docs/proposals/structured-configuration/README.md

```diff
@@ -0,0 +1,543 @@
+    // For "remove" operation, this field is ignored.
+    // +optional
+    Value apiextensionsv1.JSON `json:"value,omitempty"`
+}
+
+// YAMLPatchOperation represents a single field modification operation for YAML format.
+type YAMLPatchOperation struct {
+    // SubPath specifies the relative location within the initial FieldPath where the operation should take place.
+    // The path uses RFC 6901 for navigating into nested structures.
+    // +required
+    SubPath string `json:"subPath"`
+
+    // Operator indicates the operation on target field.
+    // Available operators are: "add", "remove", and "replace".
+    // +kubebuilder:validation:Enum=add;remove;replace
+    // +required
+    Operator OverriderOperator `json:"operator"`
+
+    // Value is the new value to set for the specified field if the operation is "add" or "replace".
+    // For "remove" operation, this field is ignored.
+    // +optional
+    Value apiextensionsv1.JSON `json:"value,omitempty"`
+}
+
+// OverriderOperator is the set of operators that can be used in an overrider.
+type OverriderOperator string
+
+// These are valid overrider operators.
+const (
+    OverriderOpAdd     OverriderOperator = "add"
+    OverriderOpRemove  OverriderOperator = "remove"
+    OverriderOpReplace OverriderOperator = "replace"
+)
+```
+
+### User usage example
+For example, consider a ConfigMap with the following data in member cluster:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example-config
+data:
+  db-config.yaml: |
+    database:
+      host: localhost
+      port: 3306
+```
+The following is OverridePolicy which uses FieldOverrider:
+```yaml
+apiVersion: karmada.io/v1alpha1
+kind: OverridePolicy
+metadata:
+  name: example-configmap-override
+spec:
+  resourceSelectors:
+    - apiVersion: v1
+      kind: ConfigMap
+      name: example-config
+  overrideRules:
+    - overriders:
+        fieldOverrider:
+          - fieldPath: /data/db-config.yaml
+            yaml:
+              - subPath: /database/host
+                operator: replace
+                value: "remote-db.example.com"
+              - subPath: /database/port
+                operator: replace
+                value: "3307"
+```
+After we apply this policy, we can modify the db-config.yaml
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example-config
+data:
+  db-config.yaml: |
+    database:
+      host: remote-db.example.com
+      port: 3307
+```
+
+### YAML Implementation
+
+We choose Plan1.
+(1) Plan1 is easy to implmenet by converting YAML directly to JSON, while Plan2 has to encapsulate ytt's grammar as if it is JSON patch.
+(2) Plan1 and Plan2 both have the same issues illustrated in the following.
+
+#### Plan 1: directly convert to and back from JSON (chosen)
+If YAML directly converts to JSON and convert back with ("sigs.k8s.io/yaml"), it will lose some data type information.
```

Comment:
```suggestion
If YAML is directly converted to JSON and then converted back using the 'sigs.k8s.io/yaml' package, it will lose some data type information.
```